### PR TITLE
build: l2 factory

### PR DESCRIPTION
**Pull request by Schlagonia**
- Use a factory that is deployed with "Nicks" method that the Bridge calls from the L1 to deploy L2 contracts

### CODE REVIEW

File: src/L1Deployer.sol [hunk 1]
```diff
@@ -195,6 +201,35 @@ contract L1Deployer is RoleManager {
...
+
+        require(_l2Admin != address(0), "ZERO ADDRESS");
+
+        address _l2Factory = getPositionHolder(L2_FACTORY);
+
+        polygonZkEVMBridge.bridgeMessage(
+            _rollupID,
+            _l2Factory,
+            true,
+            abi.encode(_l2Admin, _l2RiskManager, _l2EscrowManager)
```
Comment (fp-crypto) [inline src/L1Deployer.sol:229]: Not required, but maybe make this a struct to make the fields clear on both sides.

File: src/L2Factory.sol [hunk 2]
```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.20;
+
+import {L2Deployer} from "./L2Deployer.sol";
+import {L2Token} from "@zkevm-stb/L2Token.sol";
+import {L2Escrow} from "@zkevm-stb/L2Escrow.sol";
+import {L2TokenConverter} from "@zkevm-stb/L2TokenConverter.sol";
+
+/// @title Factory for deployment of L2 Contracts.
+contract L2Factory {
```
Comment (fp-crypto) [inline src/L2Factory.sol:10]: How do we make this contract deployed to the same address on all chains?

**Review by fp-crypto — COMMENTED** [review]
Looks good! Just a question and optional feedback. Happy we found a pattern that works.

File: src/L2Factory.sol (hunk 2 above)
Comment (Schlagonia) [reply src/L2Factory.sol:10]: This is the one that would be done through Nicks Method

Requires that the L1 Deployer and bridge address are the same though

File: src/L1Deployer.sol (hunk 1 above)
Comment (Schlagonia) [reply src/L1Deployer.sol:229]: hmm could do that for the new escrow message too

Do we trust the encoder to properly handle structs?

Comment (Schlagonia) [reply src/L1Deployer.sol:229]: added to the escrow deployment tx https://github.com/yearn/yearn-stb/commit/3e3fea05bd609fbff7a528e33bf44c1c33803796

**Comment by Schlagonia** [comment]
closing in favor of https://github.com/yearn/yearn-stb/pull/9

**Pull request closed without merging**